R/tbl-df.r

```diff
@@ -63,19 +63,21 @@ print.tbl_df <- function(x, ..., n = NULL, width = NULL, n_extra = NULL) {
   # First, subset columns
   if (!missing(j)) {
     j <- check_names_df(j, x)
-    x <- .subset(x, j)
+    result <- .subset(x, j)
+  } else {
+    result <- x
   }
 
   # Next, subset rows
   if (!missing(i)) {
-    if (length(x) == 0) {
-      nr <- length(seq_len(nr)[i])
+    if (length(result) == 0) {
+      nr <- length(attr(x, "row.names")[i])
```

Comment:
Actually no: I thought that the closer to original code the better. There is no need to use `seq_len(nr)` with this logic. If it is fine to use `seq_len(nr)` then maybe the first variant is the best.